lib/rules/id-match.js

```diff
@@ -238,6 +238,13 @@ module.exports = {
                         }
                     }
 
+                    // For https://github.com/eslint/eslint/issues/15123
+                    if (parent.parent && parent.parent.type === "ObjectExpression") {
+                        if (checkProperties && parent.key === node && isInvalid(name)) {
+                            report(node);
+                        }
+                    }
```

Comment:
Compared to the current behavior, this change would introduce a false positive:

```js
/* eslint id-match: ["error", "^[^a]", { properties: true, onlyDeclarations: true }] */

var obj = { [a]: 5 }; // false positive on 'a'

```

In this case, `a` itself isn't a property name. It's a reference to a variable (which value will become the property name), and since it isn't a variable declaration it shouldn't be reported with `onlyDeclarations: true`.

Reply:
It should be reported with `/* eslint id-match: ["error", "^[^a]", { properties: true }] */`, right? As per current behavior - [Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGlkLW1hdGNoOiBbXCJlcnJvclwiLCBcIl5bXmFdXCIsIHsgcHJvcGVydGllczogdHJ1ZSB9XSAqL1xuXG52YXIgb2JqID0geyBbYV06IDUgfTtcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

Reply:
I see there are currently no test cases present for computed properties.

Reply:
> It should be reported with `/* eslint id-match: ["error", "^[^a]", { properties: true }] */`, right? As per current behavior - [Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGlkLW1hdGNoOiBbXCJlcnJvclwiLCBcIl5bXmFdXCIsIHsgcHJvcGVydGllczogdHJ1ZSB9XSAqL1xuXG52YXIgb2JqID0geyBbYV06IDUgfTtcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

It should be reported because the default for `onlyDeclarations` is `false`, and this isn't a declaration of `a`. With `onlyDeclarations: true`, it shouldn't be reported.

`properties` option does not apply to `[a]`, because that `a` isn't a property name.